app/helpers/pipeline_outputs_helper.rb

```diff
@@ -1,6 +1,24 @@
 module PipelineOutputsHelper
   Client = Aws::S3::Client.new
 
+  def curate_pipeline_run_display(pipeline_run)
+    return nil unless pipeline_run
+    pipeline_run_display = pipeline_run.as_json.except("version")
+    pipeline_run_display["version"] = { pipeline: select_version_aspect(pipeline_run, "idseq-pipeline"),
+                                        nt: select_version_aspect(pipeline_run, "nt_k16"),
+                                        nr: select_version_aspect(pipeline_run, "nr_rapsearch") }
+    pipeline_run_display
+  end
+
+  def select_version_aspect(pipeline_run, aspect)
+    return nil unless pipeline_run.version
+    version_hash = JSON.parse(pipeline_run.version)
+    aspect_hash = version_hash.select { |item| item["name"] == aspect }[0]
+    return nil unless aspect_hash
+    version_key = %w[nt_k16 nr_rapsearch].include?(aspect) ? "source_version" : "version"
+    aspect_hash.key?(version_key) ? aspect_hash[version_key] : nil
```

Comment:
To my reading of the Ruby spec, you can just write the last line simply as follows
```
aspect_hash.key(version_key)
```

Reply:
Hmm that doesn't actually work though